tests/providers/azure/azure_test.py

```diff
@@ -13,3 +13,410 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Tests for the azure module."""
+import typing
+import unittest
+import mock
+
+from libcloudforensics.providers.azure.internal import account, compute, common
+from libcloudforensics.providers.azure import forensics
+
+# pylint: disable=line-too-long
+with mock.patch('libcloudforensics.providers.azure.internal.common.GetCredentials') as mock_creds:
+  # pylint: enable=line-too-long
+  mock_creds.return_value = mock.Mock()
+  FAKE_ACCOUNT = account.AZAccount(
+      'fake-subscription-id'
+  )
+
+FAKE_INSTANCE = compute.AZVirtualMachine(
+    FAKE_ACCOUNT,
+    '/a/b/c/fake-resource-group/fake-vm-name',
+    'fake-vm-name',
+    'fake-region',
+    ['fake-zone']
+)
+
+FAKE_DISK = compute.AZDisk(
+    FAKE_ACCOUNT,
+    '/a/b/c/fake-resource-group/fake-disk-name',
+    'fake-disk-name',
+    'fake-region',
+    ['fake-zone']
+)
+
+FAKE_BOOT_DISK = compute.AZDisk(
+    FAKE_ACCOUNT,
+    '/a/b/c/fake-resource-group/fake-boot-disk-name',
+    'fake-boot-disk-name',
+    'fake-region',
+    ['fake-zone']
+)
+
+FAKE_SNAPSHOT = compute.AZSnapshot(
+    FAKE_ACCOUNT,
+    '/a/b/c/fake-resource-group/fake_snapshot_name',
+    'fake_snapshot_name',
+    'fake-region',
+    FAKE_DISK
+)
+
+MOCK_INSTANCE = mock.Mock(
+    id='/a/b/c/fake-resource-group/fake-vm-name',
+    location='fake-region',
+    zones=['fake-zone']
+)
+MOCK_INSTANCE.name = 'fake-vm-name'
+MOCK_REQUEST_INSTANCES = [[MOCK_INSTANCE]]
+MOCK_LIST_INSTANCES = {
+    'fake-vm-name': FAKE_INSTANCE
+}
+
+MOCK_DISK = mock.Mock(
+    id='/a/b/c/fake-resource-group/fake-disk-name',
+    location='fake-region',
+    zones=['fake-zone']
+)
+MOCK_DISK.name = 'fake-disk-name'
+
+MOCK_BOOT_DISK = mock.Mock(
+    id='/a/b/c/fake-resource-group/fake-boot-disk-name',
+    location='fake-region',
+    zones=['fake-zone']
+)
+MOCK_BOOT_DISK.name = 'fake-boot-disk-name'
+
+MOCK_DISK_COPY = mock.Mock(
+    id='/a/b/c/fake-resource-group/fake_snapshot_name_f4c186ac_copy',
+    location='fake-region',
+    zones=['fake-zone']
+)
+MOCK_DISK_COPY.name = 'fake_snapshot_name_f4c186ac_copy'
```

Comment:
Because:

```
>>> mock = Mock(name='foo')
>>> mock.name
<Mock name='foo.name' id='4434041544'>

>>> mock = Mock()
>>> mock.name = 'foo'
>>> mock.name
'foo'
```

Reply:
I was not expecting that. Can you add a comment (on this line only) saying that you need to manually assigne the attribute for it not to be a `Mock` object?

Reply:
Added!